process/transactions/txGrouper.go

```diff
@@ -0,0 +1,235 @@
+		}
+
+		tx, okCast := txHandler.(*transaction.Transaction)
+		if !okCast {
+			continue
+		}
+
+		dbTx := tg.txBuilder.buildTransaction(tx, txHash, mbHash, mb, header, mbStatus)
+		addToAlteredAddresses(dbTx, alteredAddresses, mb, tg.selfShardID, false)
+		if tg.shouldIndex(mb.ReceiverShardID) {
+			transactions[string(txHash)] = dbTx
+		}
+	}
+
+	return transactions
+}
+
+func (tg *txGrouper) groupRewardsTxs(
+	mb *block.MiniBlock,
+	header nodeData.HeaderHandler,
+	txs map[string]nodeData.TransactionHandler,
+	alteredAddresses map[string]*data.AlteredAccount,
+) map[string]*data.Transaction {
+	rewardsTxs := make(map[string]*data.Transaction)
+	mbHash, err := core.CalculateHash(tg.marshalizer, tg.hasher, mb)
+	if err != nil {
+		log.Warn("txGrouper.groupRewardsTxs cannot calculate miniblock hash", "error", err)
+		return nil
+	}
+
+	mbStatus := computeStatus(tg.selfShardID, mb.ReceiverShardID)
+	for _, txHash := range mb.TxHashes {
+		txHandler, okGet := txs[string(txHash)]
+		if !okGet {
+			continue
+		}
+
+		rtx, okCast := txHandler.(*rewardTx.RewardTx)
+		if !okCast {
+			continue
+		}
+
+		dbTx := tg.txBuilder.buildRewardTransaction(rtx, txHash, mbHash, mb, header, mbStatus)
+		addToAlteredAddresses(dbTx, alteredAddresses, mb, tg.selfShardID, true)
+		if tg.shouldIndex(mb.ReceiverShardID) {
+			rewardsTxs[string(txHash)] = dbTx
+		}
+	}
+
+	return rewardsTxs
+}
+
+func (tg *txGrouper) groupInvalidTxs(
+	mb *block.MiniBlock,
+	header nodeData.HeaderHandler,
+	txs map[string]nodeData.TransactionHandler,
+	alteredAddresses map[string]*data.AlteredAccount,
+) map[string]*data.Transaction {
+	transactions := make(map[string]*data.Transaction)
+	mbHash, err := core.CalculateHash(tg.marshalizer, tg.hasher, mb)
+	if err != nil {
+		log.Warn("txGrouper.groupInvalidTxs cannot calculate miniblock hash", "error", err)
+		return nil
+	}
+
+	for _, txHash := range mb.TxHashes {
+		txHandler, okGet := txs[string(txHash)]
+		if !okGet {
+			continue
+		}
+
+		tx, okCast := txHandler.(*transaction.Transaction)
+		if !okCast {
+			continue
+		}
+
+		dbTx := tg.txBuilder.buildTransaction(tx, txHash, mbHash, mb, header, transaction.TxStatusInvalid.String())
+		addToAlteredAddresses(dbTx, alteredAddresses, mb, tg.selfShardID, false)
+
+		dbTx.GasUsed = dbTx.GasLimit
+		fee := tg.txBuilder.txFeeCalculator.ComputeTxFeeBasedOnGasUsed(tx, dbTx.GasUsed)
+		dbTx.Fee = fee.String()
+
+		transactions[string(txHash)] = dbTx
+	}
+
+	return transactions
+}
+
+func (tg *txGrouper) shouldIndex(destinationShardID uint32) bool {
+	if !tg.isInImportMode {
+		return true
+	}
+
+	return tg.selfShardID == destinationShardID
+}
+
+func (tg *txGrouper) groupReceipts(header nodeData.HeaderHandler, txPool map[string]nodeData.TransactionHandler) []*data.Receipt {
+	receipts := make(map[string]*receipt.Receipt)
+	for hash, tx := range txPool {
+		rec, ok := tx.(*receipt.Receipt)
+		if !ok {
+			continue
+		}
+
+		receipts[hash] = rec
+	}
+
+	dbReceipts := make([]*data.Receipt, 0)
```

Comment:
can extract this in a function similar to `convertMapTxsToSlice`

Reply:
refactored this method.